pymc/model_graph.py

```diff
@@ -163,7 +178,21 @@ def _make_node(self, var_name, graph, *, nx=False, cluster=False, formatting: st
                 shape = "ellipse"
                 style = None
             symbol = v.owner.op.__class__.__name__
-            if symbol.endswith("RV"):
+            if symbol == "MarginalMixtureRV":
```

Comment:
This hard-coded logic makes it impossible to customize the printed name of distributions not implemented directly in PyMC (e.g., from pymc-experimental, or third-party packages). I see two more flexible alternatives:

1. Use a dispatch based system on the `Op`
2. The `Op` can eagerly define it's specialized name when it is created. So the `init` method of `Mixture` could do this logic and override the default `self._print_name`.

I think 2. is fine for this niche application.

Reply:
We could use a dispatch system, but I believe that this would entail refactoring some parts of this file, notably `_make_node`. Should we stick with 2. for now?

What do you mean by specialized name and `init` method? I'm not sure to catch exactly what you mean here

Reply:
It wouldn't require much changes to this file. Basically, we can use the `_print_name` attribute in the `Op` instead of `__class__.__name__[:-2]`.  Those `_print_name`s are static for vanilla `RandomVariable`s but can be customized at creation time by the `SymbolicRandomVariable`s when they are initialized.

The only issue is that the `_print_name` of some RandomVariables is not desirable. For instance `pm.Normal.rv_op._print_name` is `N`. But that just means we don't like that as a `_print_name`, so we can subclass it in PyMC and give the long name.